sections/rich-text.liquid

```diff
@@ -20,19 +20,33 @@
 
 <div class="isolate{% unless section.settings.full_width %} page-width{% endunless %}">
   <div class="rich-text content-container color-{{ section.settings.color_scheme }} gradient{% if section.settings.full_width %} rich-text--full-width content-container--full-width{% endif %} section-{{ section.id }}-padding">
-    <div class="rich-text__blocks">
-      {%- for block in section.blocks -%}
-        {%- case block.type -%}
-          {%- when 'heading' -%}
-            <h2 class="{{ block.settings.heading_size }}" {{ block.shopify_attributes }}>{{ block.settings.heading | escape }}</h2>
-          {%- when 'text' -%}
-            <div class="rich-text__text rte" {{ block.shopify_attributes }}>{{ block.settings.text }}</div>
-          {%- when 'button' -%}
-            <a{% if block.settings.button_link == blank %} role="link" aria-disabled="true"{% else %} href="{{ block.settings.button_link }}"{% endif %} class="button{% if block.settings.button_style_secondary %} button--secondary{% else %} button--primary{% endif %}" {{ block.shopify_attributes }}>
-              {{ block.settings.button_label | escape }}
-            </a>
-        {%- endcase -%}
-      {%- endfor -%}
+    <div class="rich-text__wrapper rich-text__wrapper--{{ section.settings.desktop_content_position }}{% if section.settings.full_width %} page-width{% endif %}">
+      <div class="rich-text__blocks {{ section.settings.content_alignment }}">
+        {%- for block in section.blocks -%}
+          {%- case block.type -%}
+            {%- when 'heading' -%}
+              <h2 class="rich-text__heading rte {{ block.settings.heading_size }}" {{ block.shopify_attributes }}>
+                {{ block.settings.heading | replace: 'p>', 'span>' }}
+              </h2>
+            {%- when 'caption' -%}
```

Comment:
This is good feedback and I completely agree!

I didn't want to adjust any of the existing block element CSS in terms of margin/spacing etc. I favoured leaving this unaltered as I thought it was a very clean, _safe_, and well thought implementation for the blocks - I _think_ 🤞🏼 (_and feel free to correct me if I'm wrong_) my proposed changes in this PR is a good testament to this:

https://github.com/Shopify/dawn/blob/af58b34f7bf525ed7b606b35642a8e68e4c0232e/assets/section-rich-text.css#L47-L58

---

I think merchants would be better off utilizing the flexibility of the **section padding**. You could use two rich text sections on top of each other + adjust the top and bottom padding of each section to being the caption (or any other block) "closer" together - what are your thoughts?

<img src="https://user-images.githubusercontent.com/28404165/165604206-a6361b74-ff41-490f-8f35-ee52cc61b6ab.png" width="800" height="auto">

Reply:
Yeah, that's definitely a nice work around. Maybe not super straight forward for the unexperienced merchants but at least there is a way. 
Since this spacing seem to be the same across the theme we're probably better off not editing it.